### ISSUE TITLE
Weeks 10-16: focus on building a portfolio

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/master-reference#88]
Following on from discussions with @bradreeder and @des-des and others, I would like to suggest that the focus of the second half of the programme should be on building a portfolio, which probably means teams:
- choosing their own projects;
- making their own technological choices;
- working for two or more weeks on a single project;
- working on real-world applications;
- finding real users and doing user testing with them.

**Comment by sofer** [comment]
To follow up on this point, here is a suggestion for Weeks 10-16:
+ Week 10: Webapp week (one or more of React, PWA or Elm)
+ Weeks 11-12: Self-directed projects
+ Week 13: WebRTC week (IP Cortex)
+ Week 14: Design sprint 1: Anna Freud
+ Week 15: Design sprint 2 
+ Week 16: Design sprint 3

I am relatively confident of us being able to get an additional design sprint in and of getting 8⭐️  projects (up from 5) next time. Having work for most, but not all, graduates is divisive and I am keen for this not to happen next time. The post-course period could then look like this:
+ Week 17: Job application week: first round
+ Weeks 18-21: Graduate projects
+ Week 22: Job application week: second round

(⭐️ I know some people will be able to get work straight away and may not want to work at LLW, but we also have up to 6 people returning from Nazareth, so 8 2-person projects is actually the minimum to aim for)

**Comment by bradreeder** [comment]
@sofer 

Looks good 👍 Few questions:

1. I'm curious where webapp week comes from, who would be teaching it, and if not what format would it take? Would it be simpler if we just extended the self-directed project to 3 weeks, and gave students an option as to which of those technologies they explore?
2. I'd be careful to label week 13 as 'WebRTC' week, as IP Cortex might not necessarily be willing to run their workshops at all of our schools every curriculum? We could call it employer's week, and fill up other days with workshops on preparing for interviews, CVs, linkedin, etc, so students feel work-ready?
3. We need to account for curriculum planning time. We used 4 days this time and I think this is a minimum (1 day to collect feedback and assign weeks. 3 days to plan weeks.)

**Comment by sofer** [comment]
@bradreeder 

1. Good shout. Might be worth thinking about how and when to point them at relevant resources (possibly a good exercise for reading week?)
2. Another good point. I agree about *Employers' week*.
3. Curriculum planning. One day per week in Weeks 10-13, perhaps?

**Comment by iteles** [comment]
@bradreeder Was it more like 5-6 days this time round? 
Did we not do 1 day before Christmas (this may be incorrect)?

dwyl was supposed to run weeks 11 and 12 but ended up not being able to do what we had planned for multiple reasons (https://github.com/foundersandcoders/master-reference/issues/66 for example) including 3 days taken out of week 11 and 2 days taken out of week 12 for curriculum planning. I switched off after week 11, so that may not have happened in week 12 but that was the last I heard 👍

**Comment by bradreeder** [comment]
@iteles We did one day before the break (collecting curriculum feedback and assigning people to weeks), then one day of week 11, and two days of week 12.

**Comment by sofer** [comment]
@esraajb just made the point that although he spent a fair bit of time learning React during the reading weeks, it was only after Pete from JustGiving did his React workshop that it fell into place for him. So, perhaps we could give people some reading to during the break and then they could come straight back to a workshop or two on the same topics and then start on their projects.

**Comment by esraajb** [comment]
Adding to that:
The workshop with Pete(from justgiving) was very helpful. he just talked about the little things that are so confusing and less clear in the process of learning React.
I think bringing someone to the space to do 1 or 2 workshops for React (or elm, but i think it should be only one framework) is very useful.
I guess must of the students take the reading week as a good opportunity to learn things that can improve their chances to get a job, and they will probably learn React, so it makes a lot of sense to do this workshops right after the break to keep the students focused.

**Comment by sofer** [comment]
Revised proposal:

+ Week 10-12: employer/charity design sprint 1 + 2-week development sprint + workshops on Mondays + course planning on Tuesdays;
+ Week 13: employer/charity design sprint 2
+ Week 14: employer/charity design sprint 3
+ Week 15: employer/charity design sprint 4 
+ Week 16: employer/charity design sprint 5

The first sprint client might typically be a new potential partner who we are trying to build a relationship with, but who has not yet got the budget for graduate projects. The other sprints would be with partners who have committed to at least one graduate project and would generally be on some sort of tech for good theme.

Rationale:
+ the design sprints are great for learning, for enjoyment, for employer contacts and for impact;
+ but people would like more time to build a real product;
+ So they can keep learning the full stack;
+ and experiment with new libraries and frameworks;
+ and have a good portfolio project;
+ and it's nice to have a real client project where there isn't the pressure of being paid to deliver.
+ and I want a way to get new partners on board, who may not be ready to pay for graduate projects;

As discussed with @emilyb7 @skibinska @tbtommyb @msachi @lucymonie @esraajb @RhodesPeter @stevehopkinson @jsms90 @Jwhiles @SavageWilliam

**Comment by Jwhiles** [comment]
@sofer I am somewhat concerned that 6 weeks of solid design sprints might kill everyone. Is there anything we could break it up with in the middle? I think there is some value in having a relatively self directed week. 
Maybe week 14 could basically be 'choose a technology and make something cool of your own devising', interspersed with hiring workshops.

**Comment by lucymonie** [comment]
I agree with @Jwhiles and I really valued the functional programming section of the course. I would suggest:
week 10: FP, redux, game of life
week 11-12: design sprints
week 13: react 
week 15-16: design sprints

I take @esraajb's point about react but I spent my reading week on quite different things - I don't think you can count on everyone to do this unless you set it as homework. React does increasingly seem like the one thing between us and our first job.

**Comment by des-des** [comment]
@sofer I think it would be good to have build sprints either side of a day of user testing / sprint review. As it is hard to teach students about agile without this.

**Comment by jsms90** [comment]
So far we seem to have 3 distinct suggestions in this discussion, but only [one proposal](https://github.com/foundersandcoders/london-programme/issues/167#issuecomment-277462809) that has been put forward for a decision at the next curriculum committee meeting.

The history of this discussion is now spread across numerous issues, but this needs to be decided upon before FAC10 starts. I suggest that anyone who feels strongly about their idea as an alternative, raise a counter-proposal for tomorrow's meeting?

So far we have:

1) [Curriculum Committee Proposal](https://github.com/foundersandcoders/london-programme/issues/167#issuecomment-277462809) - the first one outlined in the comments above: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-277449488
+ What is the difference between week 10 and weeks 11-12?
+ Is webapp week structured like the curriculum? Or is it another self-directed project, but with a focus on a particular technology?
+ What does "self-directed project" mean in practice? Is this structured like weeks 11-12 were for FAC9?

2) @sofer 's second suggestion: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-278662848
+ What would the learning outcomes of weeks 10-12 be? It seems like a confusing amalgamation of every week that FAC9 has had over the second half of the course
+ I agree with @Jwhiles that I think we would drop down after 5-8 weeks of sprints with different clients
+ @sofer what happened to https://github.com/foundersandcoders/master-reference/issues/14#issuecomment-273225100:
> On @jsms90's point about it becoming "entirely possible that we will need all of weeks 11-15 for design sprints in future...", I agree that longer term that might be the case, although I suspect that if this happens, it will happen gradually...

3) @lucymonie 's suggestion: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-278662848
+ Would react/redux/FP weeks be structured like curriculum weeks? Or like FAC9's week on Redux / FP?

I find it difficult to properly decide upon any of these plans, without understanding what the learning outcomes and/or structure of any of these weeks actually entail.

**Comment by jsms90** [comment]
To be honest, I find it difficult to properly decide upon any of the suggestions above, without knowing what the learning outcomes and/or structure of any of these weeks would be.

So I have outlined my questions below:

### [Curriculum Committee Proposal](https://github.com/foundersandcoders/london-programme/issues/167#issuecomment-277462809) - this is also the first one outlined in the comments above: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-277449488
+ What is the difference between week 10 and weeks 11-12?
+ Is webapp week structured like the curriculum? Or is it another self-directed project, but with a focus on a particular technology?
+ What does "self-directed project" mean in practice? Is this structured like weeks 11-12 were for FAC9?

### @sofer 's second suggestion: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-278662848
+ What would the learning outcomes of weeks 10-12 be? It seems like a confusing amalgamation of every week that FAC9 has had over the second half of the course
+ I agree with @Jwhiles that I think we would drop down after 5-8 weeks of sprints with different clients
+ @sofer what happened to https://github.com/foundersandcoders/master-reference/issues/14#issuecomment-273225100:
> On @jsms90's point about it becoming "entirely possible that we will need all of weeks 11-15 for design sprints in future...", I agree that longer term that might be the case, although I suspect that if this happens, it will happen gradually...

### @lucymonie 's suggestion: https://github.com/foundersandcoders/master-reference/issues/88#issuecomment-278662848
+ Would react/redux/FP weeks be structured like curriculum weeks? Or like FAC9's week on Redux / FP?

**Comment by jsms90** [comment]
So far we seem to have 3 distinct outlined above, but only [one proposal](https://github.com/foundersandcoders/london-programme/issues/167#issuecomment-277462809) that has been put forward for a decision at the next Curriculum Committee meeting.

Normally, I would say that this dicussion is clearly not over, and therefore we are not ready for a proposal. But there have been long discussions about that are spread across numerous issues, and this needs to be decided upon before FAC10 starts.

**I suggest that anyone who feels strongly about their idea as an alternative, raise a counter-proposal for tomorrow's meeting?**

**Comment by bradreeder** [comment]
@jsms90 A comprehensive proposal for weeks 10-12, in the form of an internal agile sprint/ group project, already exists, and is in this hackpad: https://hackpad.com/product-build-curriculum-proposal-ORrmQ2jqeN2

It could easily be adapted for students to be working with real clients if that was decided. This is what me and @des-des are planning to present for 10-12 in the curriculum committee meeting as a starting point for these weeks -- as it ties together all the discussions that have happened around these weeks already. (It's hyperlinked in the agenda item.)

-------------------------------

Something we need to keep in mind @lucymonie is how under-resourced this curriculum is. It would be lovely to have weeks on FP, React, etc, but whose going to prepare the material to a high standard? Realistically the more thinly you spread yourself the lower the quality. We need to focus on not introducing more taught material until what we have is tight -- that's a big reason I've pushed for minimising the taught part down to 8 weeks. 

Internal group projects and design sprints are things I'm confident provide value to students (in terms of building a portfolio & client experience) and are things we can do well with the resources, connections and experience we have. They generate employer connections for FAC, and help us to find paid work for our students. It works as a model and is easy to implement until we have the resources to be more ambitious.

All I'd add is that I don't think we *have* to fill up 16 weeks. If you cut off at 14-15 weeks:

* Staff (like me) get a much needed chance to rest, and come to the next curriculum refreshed.
* We have ample time set-aside for curriculum planning.
* The space is still available for students to self-organise in learning new technologies together (like React) if they so choose.
* We still have room to schedule in useful employer talks, preparing for work, etc.

**Comment by lucymonie** [comment]
That's such a good point. I'm just wondering whether there may be a value in offering them their choice of topic for certain weeks, and for them to actually assemble the materials - maybe it's good practice for the mentoring to come?

**Comment by jsms90** [comment]
One other thing that doesn't really seem present in any of the proposed structures for these 8 weeks is curriculum planning. Realistically we need more than 3 days, if we are going to see real iterative improvements on the curriculum with each cohort.

We're providing post-FAC opportunities for over half of the cohort already through "employer week" and the client sprints. And @sofer has been developing more and more relationships with employers. Do we need another client sprint? We would also be providing opportunities for students to study new technologies (e.g. React / Elm) in weeks 10-13.

But the one thing that we're still missing is enough time for curriculum planning. How about week 16?

**Comment by sohilpandya** [comment]
Would like to to echo @jsms90 in terms of design sprints. I think it'll be a shame not to have for all the reasons mentioned by Jen. 

How are the students meant to cope with design sprints in week  13-16 if they don't have a clue on how it works?

**Comment by des-des** [comment]
People interested who were not present in the london-curriculum meeting. This was passed: https://github.com/foundersandcoders/london-curriculum/issues/7

**Comment by shiryz** [comment]
Will close this issue as the curriculum has deviated from what was recorded here, we now have a solid structure for weeks 10-16. If anyone feels the issue is still relevant please reopen.
